src/SimplePowerBIReportEmbed/SimplePowerBIReportEmbed.js

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useEffect, useMemo, useState } from 'react';
+import { PowerBIEmbed } from 'powerbi-client-react';
+import * as PropTypes from 'prop-types';
+import AccessTimeIcon from '@material-ui/icons/AccessTime';
+import DashboardPlaceholder from '../Dashboard/components';
+import { PowerBIUtils } from '@cosmotech/core';
+import { useTranslation } from 'react-i18next';
+import { makeStyles } from '@material-ui/core';
+
+const useStyles = makeStyles(theme => ({
+  divContainer: {
+    height: '100%',
+    width: '100%'
+  }
+}));
+
+const SimplePowerBIReportEmbed = ({ index, reports, reportConfiguration, scenario, scenarioList, lang, downloadLogsFile }) => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const { reportId, settings, staticFilters, dynamicFilters, pageName } = reportConfiguration[index];
+
+  const [embedConfig, setEmbedConfig] = useState({
+    type: 'report',
+    id: reportId,
+    embedUrl: '',
+    accessToken: '',
+    tokenType: 1 // 1 Embed or 0 Aad
+  });
+
+  let noRun = false, scenarioState = 'Created', runInProgress = false, hasError = false, isReady = false, noScenario = true;
+  let additionalFilters = [];
+
+  if (scenario) {
+    const scenarioDTO = useMemo( () => PowerBIUtils.constructScenarioDTO(scenario, scenarioList), [scenario,scenarioList]);
+    additionalFilters = useMemo(() => PowerBIUtils.constructDynamicFilters(dynamicFilters, scenarioDTO), [dynamicFilters, scenarioDTO]);
+    noScenario = scenarioDTO === null;
+    scenarioState = scenarioDTO.state;
+    noRun = scenarioState === 'Created' || scenarioState === null;
+    runInProgress = scenarioState === 'Running';
+    hasError = scenarioState === 'Failed';
+    isReady = (scenarioState === undefined || scenarioState === 'Successful') && !noScenario;
+  }
+
+  useEffect(() => {
+    const newConfig = {
+      type: 'report',
+      id: reportId,
+      tokenType: 1,
+      embedUrl: reports.data?.embedUrl[reportId]?.embedUrl,
+      accessToken: reports.data?.accessToken
+    };
+
+    if (pageName !== undefined && pageName[lang] !== undefined) {
+      newConfig.pageName = pageName[lang];
+    }
+
+    if (settings !== undefined) {
+      newConfig.settings = settings;
+    }
+
+    if (staticFilters !== undefined || additionalFilters !== undefined) {
+      let filters = [];
+      if (staticFilters !== undefined) {
+        filters = filters.concat(staticFilters);
+      }
+      if (additionalFilters !== undefined) {
+        filters = filters.concat(additionalFilters);
+      }
+      newConfig.filters = filters;
+    }
+    setEmbedConfig(newConfig);
```

Comment:
Could we create one or more functions to make the useEffect block more compact ?

Reply:
:+1: